eth2/utils/eth2_keystore/tests/json.rs

```diff
@@ -0,0 +1,674 @@
+}
+
+#[test]
+fn pbkdf2_additional_parameter() {
+    let vector = r#"
+            {
+            "crypto": {
+                "kdf": {
+                    "function": "pbkdf2",
+                    "params": {
+                        "dklen": 32,
+                        "c": 262144,
+                        "prf": "hmac-sha256",
+                        "salt": "d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
+                        "cats": 42
+                    },
+                    "message": ""
+                },
+                "checksum": {
+                    "function": "sha256",
+                    "params": {},
+                    "message": "18b148af8e52920318084560fd766f9d09587b4915258dec0676cba5b0da09d8"
+                },
+                "cipher": {
+                    "function": "aes-128-ctr",
+                    "params": {
+                        "iv": "264daa3f303d7259501c93d997d84fe6"
+                    },
+                    "message": "a9249e0ca7315836356e4c7440361ff22b9fe71e2e2ed34fc1eb03976924ed48"
+                }
+            },
+            "pubkey": "9612d7a727c9d0a22e185a1c768478dfe919cada9266988cb32359c11f2b7b27f4ae4040902382ae2910c15e2b420d07",
+            "path": "m/12381/60/0/0",
+            "uuid": "64625def-3331-4eea-ab6f-782f3ed16a83",
+            "version": 4
+        }
+        "#;
+
+    match Keystore::from_json_str(&vector) {
+        Err(Error::InvalidJson(_)) => {}
+        _ => panic!("expected invalid json error"),
+    }
+}
+
+#[test]
+fn pbkdf2_missing_parameter() {
+    let vector = r#"
+            {
+            "crypto": {
+                "kdf": {
+                    "function": "pbkdf2",
+                    "params": {
+                        "c": 262144,
+                        "prf": "hmac-sha256",
+                        "salt": "d4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3"
+                    },
+                    "message": ""
+                },
+                "checksum": {
+                    "function": "sha256",
+                    "params": {},
+                    "message": "18b148af8e52920318084560fd766f9d09587b4915258dec0676cba5b0da09d8"
+                },
+                "cipher": {
+                    "function": "aes-128-ctr",
+                    "params": {
+                        "iv": "264daa3f303d7259501c93d997d84fe6"
+                    },
+                    "message": "a9249e0ca7315836356e4c7440361ff22b9fe71e2e2ed34fc1eb03976924ed48"
+                }
+            },
+            "pubkey": "9612d7a727c9d0a22e185a1c768478dfe919cada9266988cb32359c11f2b7b27f4ae4040902382ae2910c15e2b420d07",
+            "path": "m/12381/60/0/0",
+            "uuid": "64625def-3331-4eea-ab6f-782f3ed16a83",
+            "version": 4
+        }
+        "#;
+
+    match Keystore::from_json_str(&vector) {
+        Err(Error::InvalidJson(_)) => {}
+        _ => panic!("expected invalid json error"),
+    }
+}
```

Comment:
Should we add test if there are extra fields being used? Currently including an extra field in any of the following objects will result in valid deserialisation:
* cypto
* kdf
* checksum
* cipher
* top level (e.g. before "crypto")

E.g. the following is valid
```json
{
"crypto": {
    "extra_field": "",
    "kdf": { ... },
...
}
```

Reply:
All of these have been covered, thank you!